### PR TITLE
Route data on the HttpContext scenario

### LINKED ISSUE #9558: Route data on the HttpContext scenario
For 3.0 in the *URL matching* section (insert 2nd para):

Cross-ref: https://github.com/aspnet/AspNetCore.Docs/pull/9527#discussion_r232505921

> When a Routing Middleware executes, it sets an endpoint (\`Endpoint\`) and route values to a feature on the \<xref:Microsoft.AspNetCore.Http.HttpContext>. Middleware running after the Routing Middleware can see the endpoint and can take actions based on the middleware. For example, an authorization middleware can interrogate the endpoint's metadata collection for an authorization policy. After all of the middleware in the request processing pipeline is executed, the selected endpoint's delegate is invoked.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 32374077-e02e-d6a6-1cea-05e58447b7cd
* Version Independent ID: 86aeab13-1c25-9fc1-4e56-0390cb40c242
* Content: [Routing in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-2.1)
* Content Source: [aspnetcore/fundamentals/routing.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/routing.md)
* Product: **aspnet-core**
* GitHub Login: @ardalis
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #9558 

Per https://github.com/aspnet/AspNetCore.Docs/pull/9527#discussion_r232505921 and the fact that we have doc versioning now for 3.0+, we can get this in now.

### CODE REVIEW

File: aspnetcore/fundamentals/routing.md [hunk 1]
```diff
@@ -135,6 +135,16 @@ Routing is connected to the [middleware](xref:fundamentals/middleware/index) pip
 
 URL matching is the process by which routing dispatches an incoming request to an *endpoint*. This process is based on data in the URL path but can be extended to consider any data in the request. The ability to dispatch requests to separate handlers is key to scaling the size and complexity of an app.
 
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
+
+When a Routing Middleware executes, it sets an endpoint (`Endpoint`) and route values to a feature on the <xref:Microsoft.AspNetCore.Http.HttpContext>. Middleware running after the Routing Middleware can see the endpoint and can take actions based on the middleware. For example, an Authorization Middleware can interrogate the endpoint's metadata collection for an authorization policy. After all of the middleware in the request processing pipeline is executed, the selected endpoint's delegate is invoked.
```
Comment (JamesNK) [inline aspnetcore/fundamentals/routing.md]: I think this should be expanded to include how to get the endpoint/route values after they have been set.

* HttpContext.GetEndpoint gets the endpoint - https://github.com/aspnet/AspNetCore/blob/f45250ce5e423dad083fe99967a4d65782c5301e/src/Http/Http.Abstractions/src/Routing/EndpointHttpContextExtensions.cs#L19
* HttpRequest.RouteValues gets the route values - https://github.com/aspnet/AspNetCore/blob/f45250ce5e423dad083fe99967a4d65782c5301e/src/Http/Http.Abstractions/src/HttpRequest.cs#L132

**Comment by guardrex** [comment]
@JamesNK I can't link those to API docs until 3.0 RTM. If that (or something along those lines) works, I'll open a follow-up issue for the links.

File: aspnetcore/fundamentals/routing.md [hunk 2]
```diff
@@ -135,6 +135,21 @@ Routing is connected to the [middleware](xref:fundamentals/middleware/index) pip
 
 URL matching is the process by which routing dispatches an incoming request to an *endpoint*. This process is based on data in the URL path but can be extended to consider any data in the request. The ability to dispatch requests to separate handlers is key to scaling the size and complexity of an app.
 
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
+
+When a Routing Middleware executes, it sets an endpoint (`Endpoint`) and route values to a feature on the <xref:Microsoft.AspNetCore.Http.HttpContext>. For the current request:
+
+* `HttpContext.GetEndpoint` gets the endpoint.
```
Comment (JamesNK) [inline aspnetcore/fundamentals/routing.md]:
```suggestion
* `HttpContext.GetEndpoint()` gets the endpoint.
```

Comment (guardrex) [reply aspnetcore/fundamentals/routing.md]: 🤔 hum ... we don't ordinarily show paren for methods. How about this ...

```suggestion
* Calling `HttpContext.GetEndpoint` gets the endpoint.
```

Comment (guardrex) [reply aspnetcore/fundamentals/routing.md]: Yeah ... let's do that. We aren't going to get the paren after we use the `xref` link later anyway.

btw- I made a note on ...

**[RTM] Remove version context from Routing** #9552

... to address the links. I (or someone else) will take care of it.

**Pull request merged**